### PR TITLE
Danny Ramos

### PR DESCRIPTION
**Pull request by muydanny**
:/

### CODE REVIEW

**Comment by timomitchel** [comment]
Would have loved to see commits throughout, and the tests committed first as the directions called for.

File: lib/event.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      names << food_truck.name
+    end
+    names
+  end
+  # `food_trucks_that_sell` that takes an
+  # argument of an item represented as a String.
+  # It will return a list
+  # of FoodTrucks that have that item in stock.
+
+  def food_trucks_that_sell(desired_item)
```
Comment (timomitchel) [inline lib/event.rb:27]: Nice job with this method! This works, but you can save yourself some time and some lines with a combination of `find_all` and `includes?`:
```
@food_trucks.find_all {|food_truck| food_truck.inventory.include?(item)}
```

File: lib/event.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def food_truck_names
+    names = []
+    @food_trucks.each do |food_truck|
+      names << food_truck.name
+    end
+    names
+  end
+  # `food_trucks_that_sell` that takes an
```
Comment (timomitchel) [inline lib/event.rb:22]: I love this pseudo coding! It worked!

File: lib/food_truck.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  def stock(item, quantity)
+    @inventory[item] = quantity
+  end
+
+  # the sum of all their items'
+  # price * quantity.
+
+  def potential_revenue
+    @inventory.sum do |inventory|
```
Comment (timomitchel) [inline lib/food_truck.rb:26]: You should be splitting this hash by key and value pairs in the block variable here:
```
 @inventory.sum do |item, quantity|
item.price * quantity
end
```

File: lib/food_truck.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+class FoodTruck
+
+  attr_reader :name,
+              :inventory
+
+  def initialize(name)
+    @name = name
+    @inventory = {}
```
Comment (timomitchel) [inline lib/food_truck.rb:8]: We were hoping to see this hash initialized with a default value of 0 based on the interaction patter. You can do this two ways:
```
Hash.new(0)
```
Or after creating the hash, you could assign the default value by calling .default on the hash
```
@inventory.default = 0

File: lib/food_truck.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    @default_stock = 0
+  end
+
+  def check_stock(item)
+    @inventory.sum do |item|
```
Comment (timomitchel) [inline lib/food_truck.rb:13]: This method needed to return the value for the item key in our inventory hash:
```
@inventory[item]
```
which would return 0 if the key did not exist and the default value of the hash was 0

File: lib/food_truck.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      item[1] += @default_stock
+    end
+  end
+
+  def stock(item, quantity)
+    @inventory[item] = quantity
```
Comment (timomitchel) [inline lib/food_truck.rb:19]: We were looking for something different here:
```
if @inventory.has_key?(item)
@inventory[item] += quantity
else
@inventory[item] = quantity
end
```
When the condition returns true, we add to the current value of item, when it's false, we add the item key to the hash and assign it a quantity.

File: lib/item.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader :name
+
+
+  def initialize(item_info)
+    @name = item_info[:name]
+    @price = item_info[:price]
+  end
+
+  def price
+    @price.gsub(/[^\d\.]/, '').to_f
```
Comment (timomitchel) [inline lib/item.rb:12]: Oh you fancy with your regex ay?

File: lib/item.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class Item
+
+  attr_reader :name
+
+
+  def initialize(item_info)
+    @name = item_info[:name]
+    @price = item_info[:price]
```
Comment (timomitchel) [inline lib/item.rb:8]: It's preferred to do the conversion for `@price` right in initialize and then use and `attr_reade`r for price:
`@price =info[:price].gsub("$", "").to_f`
This way the conversion only runs once, not every time you call the #price method.

File: test/event_test.rb [hunk 9]
```diff
@@ -0,0 +1,74 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/item'
+require './lib/food_truck'
+require './lib/event'
+
+class EventTest < Minitest::Test
+  def setup
```
Comment (timomitchel) [inline test/event_test.rb:8]: Nice job utilizing a setup method here to save time.

File: test/event_test.rb [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      assert_equal "South Pearl Street Farmers Market", @event.name
+      assert_equal [], @event.food_trucks
+    end
+
+    def test_event_has_food_truck
+      @food_truck1.stock(@item1, 35)
+      @food_truck1.stock(@item2, 7)
+      @food_truck2.stock(@item4, 50)
+      @food_truck2.stock(@item3, 25)
+      @food_truck3.stock(@item1, 65)
```
Comment (timomitchel) [inline test/event_test.rb:33]: No need for the setup from 29 - 33 here. We just want to make sure we can add food trucks, but they don't need inventory for this test to work properly.

File: test/event_test.rb [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+      @event.add_food_truck(@food_truck3)
+      assert_equal [@food_truck1, @food_truck2, @food_truck3], @event.food_trucks
+    end
+
+    def test_event_can_give_food_truck_names
+      @event.add_food_truck(@food_truck1)
+      @event.add_food_truck(@food_truck2)
+      @event.add_food_truck(@food_truck3)
+      expected = ["Rocky Mountain Pies", "Ba-Nom-a-Nom", "Palisade Peach Shack"]
+      assert_equal expected, @event.food_truck_names
```
Comment (timomitchel) [inline test/event_test.rb:45]: Nice job here.

File: test/event_test.rb [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+      @food_truck1.stock(@item1, 35)
+      @food_truck1.stock(@item2, 7)
+      @food_truck2.stock(@item4, 50)
+      @food_truck2.stock(@item3, 25)
+      @food_truck3.stock(@item1, 65)
+      @event.add_food_truck(@food_truck1)
+      @event.add_food_truck(@food_truck2)
+      @event.add_food_truck(@food_truck3)
+      assert_equal [@food_truck1, @food_truck3], @event.food_trucks_that_sell(@item1)
+      assert_equal [@food_truck2], @event.food_trucks_that_sell(@item4)
```
Comment (timomitchel) [inline test/event_test.rb:58]: Great job robustly testing this method!

File: test/event_test.rb [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+      @food_truck1.stock(@item2, 7)
+      @food_truck2.stock(@item4, 50)
+      @food_truck2.stock(@item3, 25)
+      @food_truck3.stock(@item1, 65)
+      @food_truck3.stock(@item3, 10)
+      @event.add_food_truck(@food_truck1)
+      @event.add_food_truck(@food_truck2)
+      @event.add_food_truck(@food_truck3)
+      expected = { @item1 => {quantity => 100, food_trucks => [food_truck1, food_truck3]}}
+      assert_equal expected, @event.total_inventory
```
Comment (timomitchel) [inline test/event_test.rb:72]: The setup is incorrect here. We needed to have each of the stocked items as keys in our hash.

File: test/food_truck_test.rb [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+    def test_it_can_stock
+       @food_truck.stock(@item1, 30)
+       expected = {@item1 => 30}
+       assert_equal expected, @food_truck.inventory
+       assert_equal 30, @food_truck.check_stock(@item1)
+       @food_truck.stock(@item1, 25)
+       # assert_equal 55, @food_truck.check_stock(@item1)
+       @food_truck.stock(@item2, 12)
+       expected = {@item1 => 55, @item2 => 12}
+       # assert_equal expected, @food_truck.inventory
```
Comment (timomitchel) [inline test/food_truck_test.rb:40]: This should have caused more alarm. Always best to get a method working before moving on.

File: test/food_truck_test.rb [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+    def test_potential_revenue_of_food_truck
+      @food_truck1.stock(@item1, 35)
+      @food_truck1.stock(@item2, 7)
+      @food_truck2.stock(@item4, 50)
+      @food_truck2.stock(@item3, 25)
+      @food_truck3.stock(@item1, 65)
+      assert_equal 148.75, @food_truck1.potential_revenue
```
Comment (timomitchel) [inline test/food_truck_test.rb:49]: Nice job with this test.

File: lib/item.rb (hunk 7 above)
Comment (muydanny) [reply lib/item.rb:12]: lmaooo this was such a shot in the dark with a google search and when it worked I literally jumped out  my chair but yeah Im fancy

**Comment by muydanny** [comment]
@timomitchel Thanks Tim! Methods took me longer than I wanted but this assessment felt immensely better than previous ones. Pry really does save lives lol Although Im not quite there yet with more work and studying Im feeling good going into the last week.

**Pull request closed without merging**